plugins/legion-go2/fu-legion-go2-device.c

```diff
@@ -0,0 +1,671 @@
+/*
+ * Copyright 2025 lazro <li@shzj.cc>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+#include "config.h"
+
+#include "fu-legion-go2-device.h"
+#include "fu-legion-go2-struct.h"
+
+#define FU_LEGION_GO2_DEVICE_VID       0x17EF
+#define FU_LEGION_GO2_DEVICE_PID_BEGIN 0x61EB
+#define FU_LEGION_GO2_DEVICE_PID_END   0x61EE
+
+#define FU_LEGION_GO2_DEVICE_IO_TIMEOUT            500
+#define FU_LEGION_GO2_DEVICE_BUSY_WAIT_TIMEOUT     (60*1000)
+#define FU_LEGION_GO2_DEVICE_READ_RESPONSE_TIMEOUT 2000
+
+#define FU_LEGION_GO2_DEVICE_FW_SIGNED_LENGTH 384
+#define FU_LEGION_GO2_DEVICE_FW_ID_LENGTH     4
+#define FU_LEGION_GO2_DEVICE_FW_PACKET_LENGTH 32
+#define FU_LEGION_GO2_DEVICE_FW_REPORT_LENGTH 64
+
+
+struct _FuLegionGo2Device {
+	FuHidrawDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuLegionGo2Device, fu_legion_go2_device, FU_TYPE_HIDRAW_DEVICE)
+
+static gboolean
+fu_legion_go2_device_read_response(FuLegionGo2Device *self, guint8 main_id, guint8 sub_id, GByteArray* res)
```

Comment:
this should have a GError as a last parameter, which needs to be set wen this returns FALSE.